### PR TITLE
fix sprite rt adaptation

### PR DESCRIPTION
**Pull request by yanjifa**
Re: # https://forum.cocos.org/t/topic/155680

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/components/sprite.ts [hunk 1]
```diff
@@ -696,6 +692,10 @@ export class Sprite extends UIRenderer {
             }
             if (textureChanged) {
                 if (this.renderData) this.renderData.textureDirty = true;
+                // texture type changed, set this._instanceMaterialType to default value
+                if ((oldFrame ? oldFrame.texture instanceof RenderTexture : false) !== spriteFrame.texture instanceof RenderTexture) {
```
Comment (qiuguohua) [inline cocos/2d/components/sprite.ts]: When judging texturechange earlier, we have ensured that oldFrame.texture is not equal to spriteFrame.texture？

Comment (yanjifa) [reply cocos/2d/components/sprite.ts]: So what I'm judging here is whether the texture is changing from RenderTexture to a regular Texture, or from a regular Texture to RenderTexture.

Comment (minggo) [reply cocos/2d/components/sprite.ts]: The code looks wired. Could you please modify it to make it more clear?

Comment (qiuguohua) [reply cocos/2d/components/sprite.ts]: ok, got it

**Review by GengineJS — APPROVED** [review]

**Review by qiuguohua — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
@yanjifa could you please send to v3.8.4 as v3.8.3 will be released soon. Thanks.

**Pull request merged**